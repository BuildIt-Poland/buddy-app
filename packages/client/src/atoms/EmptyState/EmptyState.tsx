import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import DICTIONARY from './dictionary';

const useStyles = makeStyles(theme => ({
  emptyContainer: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
}));

const EmptyState = () => {
  const { emptyContainer } = useStyles();
  return (
    <Box className={emptyContainer}>
      <Typography variant='h2' component='h2'>
        {DICTIONARY.TITLE}
      </Typography>
    </Box>
  );
};

export default EmptyState;
