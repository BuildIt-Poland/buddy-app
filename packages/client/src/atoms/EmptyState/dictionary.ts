interface Dictionary {
  TITLE: string;
}

const DICTIONARY: Dictionary = {
  TITLE: `Whoops! No data found..`,
};

export default DICTIONARY;
