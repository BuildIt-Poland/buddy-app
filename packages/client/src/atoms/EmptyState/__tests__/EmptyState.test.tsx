import React from 'react';
import { create } from 'react-test-renderer';

import EmptyState from '../EmptyState';

jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('atoms/R2d2', () => 'R2d2');

describe('Component - EmptyState', () => {
  test('renders correctly', () => {
    const component = create(<EmptyState />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
