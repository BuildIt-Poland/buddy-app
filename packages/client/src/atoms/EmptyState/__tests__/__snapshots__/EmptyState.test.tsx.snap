// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - EmptyState renders correctly 1`] = `
<Box
  className="makeStyles-emptyContainer-1"
>
  <Typography
    component="h2"
    variant="h2"
  >
    Whoops! There's no data..
  </Typography>
</Box>
`;
